### PR TITLE
Improving Kalman filter

### PR DESCRIPTION
**Pull request by raphaelsaavedra**

### CODE REVIEW

File: src/kalman.jl [hunk 1]
```diff
@@ -26,67 +20,62 @@ function sqrt_kalmanfilter(model::StateSpaceModel, sqrtH::Matrix{Typ}, sqrtQ::Ma
...
             G            = qr(U').Q
             Ustar        = U*G
             U2star[t]    = Ustar[(p + 1):(p + m), 1:p]
             sqrtF[t]     = Ustar[1:p, 1:p]
 
-            # Kalman gain
-            K[t]       = U2star[t] * pinv(sqrtF[t])
-            a[t+1]     = T * a[t] + K[t] * v[t]
+            # Kalman gain and predictive state update
+            K[t]       = U2star[t]*pinv(sqrtF[t])
```
Comment (guilhermebodin) [inline src/kalman.jl]: I think all `pinv()` could be inplace `pinv.()`

File: src/kalman.jl [hunk 2]
```diff
@@ -26,67 +20,62 @@ function sqrt_kalmanfilter(model::StateSpaceModel, sqrtH::Matrix{Typ}, sqrtQ::Ma
...
 
-    # Kalman gain and steady-state Kalman gain
+    # Kalman gain
     K       = Vector{Matrix{Float64}}(undef, n)
-    Ksteady = Matrix{Float64}(undef, m, p)
-
-    # Predictive variance in steady state
-    Ps = Matrix{Float64}(undef, m, m)
 
     # Auxiliary matrices
```
Comment (guilhermebodin) [inline src/kalman.jl]: I am a bit lost in what `U` and `U2star` are used for, maybe more comments on what auxiliary matrices are used for could provide some better understanding

File: src/structures.jl [hunk 3]
```diff
@@ -87,24 +87,17 @@ Following the notation of on the book \"Time Series Analysis by State Space Meth
 
 * `a`
```
Comment (guilhermebodin) [inline src/structures.jl]: If you can, it would be great to document all the parameters properly, but this could also be in another PR

File: src/kalman.jl (hunk 2 above)
Comment (raphaelsaavedra) [reply src/kalman.jl]: This part is complicated and is something that the user shouldn't really have to worry about. I think the main downside of the latest change is that `U2star` is being returned by the filter function, so in the future if we make the future an exported, standalone function it's going to be weird. 

Conversely, maintaining `U2star` in the `FilterOutput` object wasn't nice because it's basically a useless piece of data. We can think about this later.

File: src/kalman.jl (hunk 1 above)
Comment (raphaelsaavedra) [reply src/kalman.jl]: True, changed it

File: src/structures.jl (hunk 3 above)
Comment (raphaelsaavedra) [reply src/structures.jl]: This is something that we should do in the future. I'll open an issue.
